ImmichFrame.WebApi/Models/ServerSettings.cs

```diff
@@ -1,51 +1,94 @@
-        public int? ImagesFromDays { get; set; }
-        public DateTime? ImagesFromDate { get; set; }
-        public DateTime? ImagesUntilDate { get; set; }
-        public List<Guid> Albums { get; set; } = new List<Guid>();
-        public List<Guid> ExcludedAlbums { get; set; } = new List<Guid>();
-        public List<Guid> People { get; set; } = new List<Guid>();
-        public int? Rating { get; set; }
-        public List<string> Webcalendars { get; set; } = new List<string>();
-        public int RefreshAlbumPeopleInterval { get; set; } = 12;
-        public string ImmichFrameAlbumName { get; set; } = string.Empty;
-        public string? WeatherApiKey { get; set; } = string.Empty;
-        public string? UnitSystem { get; set; } = "imperial";
-        public string? WeatherLatLong { get; set; } = "40.7128,74.0060";
-        public string Language { get; set; } = "en";
-        public string? Webhook { get; set; }
-        public string? AuthenticationSecret { get; set; }
+        public class ServerFrameSettings : IImmichFrameSettings
+        {
+            public bool DownloadImages { get; set; } = false;
+            public int RenewImagesDuration { get; set; } = 30;
+            public List<string> Webcalendars { get; set; } = new List<string>();
+            public int RefreshAlbumPeopleInterval { get; set; } = 12;
+            public string? WeatherApiKey { get; set; } = string.Empty;
+            public string? UnitSystem { get; set; } = "imperial";
+            public string? WeatherLatLong { get; set; } = "40.7128,74.0060";
+            public string Language { get; set; } = "en";
+            public string? Webhook { get; set; }
+            public string? AuthenticationSecret { get; set; }
+        }
+
+        public class ServerAccountSettings : IImmichAccountSettings
+        {
+            public string ImmichServerUrl { get; set; } = string.Empty;
+            public string ApiKey { get; set; } = string.Empty;
+            public bool ShowMemories { get; set; } = false;
+            public bool ShowFavorites { get; set; } = false;
+            public bool ShowArchived { get; set; } = false;
+
+            public int? ImagesFromDays { get; set; }
+            public DateTime? ImagesFromDate { get; set; }
+            public DateTime? ImagesUntilDate { get; set; }
+            public List<Guid> Albums { get; set; } = new List<Guid>();
+            public List<Guid> ExcludedAlbums { get; set; } = new List<Guid>();
+            public List<Guid> People { get; set; } = new List<Guid>();
+            public int? Rating { get; set; }
+
+            public string ImmichFrameAlbumName { get; set; } = string.Empty;
+        }
+
+        public IImmichFrameSettings ImmichFrameSettings { get; set; } = new ServerFrameSettings();
+        public IList<IImmichAccountSettings> Accounts { get; set; } = new List<IImmichAccountSettings>();
+
+
+        private Regex _accountSettingRegex = new(@"^Account(\d)\.(\w+)$");
```

Comment:
This is nitpicking, sorry. What do you think about using named regex groups?

```suggestion
        private Regex _accountSettingRegex = new(@"^Account(?<id>\d)\.(?<property>\w+)$");
```

Reply:
fine by me; I'll update.